File: programmers/[Summer,Winter-Coding(~2018)]/스킬트리.py
# 스킬트리

def solution(skill, skill_trees):
    answer = 0
    a = list(skill)

    for i in skill_trees:
        i = list(i)
        temp = []
        for j in range(len(i)):
            for k in range(len(a)):
                if i[j] == a[k]:
                    temp.append(i[j])
        if ''.join(temp) == skill[:len(temp)]:
            answer += 1

    return answer

File: Algorithm-Swift/Algorithm-Swift/Programmers/Summer,Winter Coding(~2018)/스킬트리.swift
//
//  스킬트리.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/05/07.
//
import Foundation

func check(_ custom: [Character], _ preceded: [Character]) -> Bool{
    var temp = [Character]()
    for i in custom{
        if preceded.contains(i){
            if preceded[(temp.count) % preceded.count] != i{
                return false   
            }
            temp.append(i)
        }
    }
    return true
}

func solution(_ skill:String, _ skill_trees:[String]) -> Int {
    var answer = 0
    let preceded = Array(skill)
    
    for i in skill_trees{
        let custom = Array(i)
        if check(custom, preceded){
            answer += 1
        }
    }
    return answer
}